x-pack/plugin/searchable-snapshots/src/test/java/org/elasticsearch/xpack/searchablesnapshots/AbstractSearchableSnapshotsTestCase.java

```diff
@@ -0,0 +1,138 @@
+    }
+
+    /**
+     * @return a new {@link CacheService} instance configured with the given cache size and cache range size settings
+     */
+    protected CacheService createCacheService(final ByteSizeValue cacheSize, final ByteSizeValue cacheRangeSize) {
+        return new CacheService(
+            AbstractSearchableSnapshotsTestCase::noOpCacheCleaner,
+            Settings.builder()
+                .put(CacheService.SNAPSHOT_CACHE_SIZE_SETTING.getKey(), cacheSize)
+                .put(CacheService.SNAPSHOT_CACHE_RANGE_SIZE_SETTING.getKey(), cacheRangeSize)
+                .build()
+        );
+    }
+
+    protected static void noOpCacheCleaner() {}
+
+    /**
+     * @return a random {@link ByteSizeValue} that can be used to set {@link CacheService#SNAPSHOT_CACHE_SIZE_SETTING}.
+     * Note that it can return a cache size of 0.
+     */
+    protected static ByteSizeValue randomCacheSize() {
+        return new ByteSizeValue(randomNonNegativeLong());
+    }
+
+    /**
+     * @return a random {@link ByteSizeValue} that can be used to set {@link CacheService#SNAPSHOT_CACHE_RANGE_SIZE_SETTING}
+     */
+    protected static ByteSizeValue randomCacheRangeSize() {
+        return new ByteSizeValue(
+            randomLongBetween(CacheService.MIN_SNAPSHOT_CACHE_RANGE_SIZE.getBytes(), CacheService.MAX_SNAPSHOT_CACHE_RANGE_SIZE.getBytes())
+        );
+    }
+
+    protected static SearchableSnapshotRecoveryState createRecoveryState() {
+        ShardRouting shardRouting = TestShardRouting.newShardRouting(
+            new ShardId(randomAlphaOfLength(10), randomAlphaOfLength(10), 0),
+            randomAlphaOfLength(10),
+            true,
+            ShardRoutingState.INITIALIZING,
+            new RecoverySource.SnapshotRecoverySource(
+                UUIDs.randomBase64UUID(),
+                new Snapshot("repo", new SnapshotId(randomAlphaOfLength(8), UUIDs.randomBase64UUID())),
+                Version.CURRENT,
+                new IndexId("some_index", UUIDs.randomBase64UUID(random()))
+            )
+        );
+        DiscoveryNode targetNode = new DiscoveryNode("local", buildNewFakeTransportAddress(), Version.CURRENT);
+        SearchableSnapshotRecoveryState recoveryState = new SearchableSnapshotRecoveryState(shardRouting, targetNode, null);
+
+        recoveryState.setStage(RecoveryState.Stage.INIT)
+            .setStage(RecoveryState.Stage.INDEX)
+            .setStage(RecoveryState.Stage.VERIFY_INDEX)
+            .setStage(RecoveryState.Stage.TRANSLOG);
+        recoveryState.getIndex().setFileDetailsComplete();
+        recoveryState.setStage(RecoveryState.Stage.FINALIZE).setStage(RecoveryState.Stage.DONE);
+
+        return recoveryState;
+    }
+
+    /**
+     * Wait for all operations on the threadpool to complete
+     */
+    protected static void assertThreadPoolNotBusy(ThreadPool threadPool) throws Exception {
+        assertBusy(() -> {
```

Comment:
The method this replaces in some spots used a 30s timeout. When I created it I just copied those 30s from previous code so I don't know if we actually need the long of a timeout here, but maybe we do and should put I back here?

Reply:
Thanks for catching this. I looked a bit at the history and I think it's preferable to use a long timeout here (mostly for testRandomReads())